docs/Components-Header.md

```diff
@@ -221,16 +222,84 @@ The `onRightPress` is an optional prop that is used to set the function to be ca
 
 ### `safeAreaEdges`
 
-The `safeAreaEdges` optional prop can be used to override the default safe area edges. By default, the header will use the `top` safe area edge. If you want to not account for the safe area edges, you can pass in `[]` to the `safeAreaEdges` prop, and it will wrap the header in a `View` component instead of a `SafeAreaView` component.
+The `safeAreaEdges` optional prop can be used to override the default safe area edges. By default, the header will use the `top` safe area edge. If you want to not account for the safe area edges, you can pass in `[]` to the `safeAreaEdges` prop.
 
 ```tsx
 <Header titleTx="header.title" safeAreaEdges={[]} />
 ```
 
-### `SafeAreaViewProps`
+## Usage
+
+The Header is a flexible component that can be integrated within your application using a few different methods:
+
+### Method 1 (recommended) - Using `navigation.setOptions()` method in your screen component or `useHeader()` hook.
+
+This method gives you the most control over your Header and co-locates the logic inside of your screen while preserving react-navigation's optimizations by keeping it outside of the screen component's render lifecycle.
+
+```tsx
+function AccountScreen(props) {
+  const { navigation } = props
+
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      headerShown: true,
+      header: () => <Header title="Hello" />,
+    })
+  }, [])
+
+  return <Screen />
+}
+```
+
+A convenience [`useHeader`](./Utils-useHeader.md) hook is provided that abstracts and cleans up some of this logic.
+
+```tsx
+function AccountScreen(props) {
+  useHeader({
+    headerShown: true,
+    header: () => <Header title="Hello" />,
+  })
```

Comment:
```suggestion
  useHeader({
    title: "Hello",
  })
```

The API of this useHeader is you just pass in the props, I believe.

Reply:
You're right, my mistake.. copy pasting stuff is hard.